customized handlebars files can not be translated

modifying views for a domain in `meshcentral-web` or `meshcentral-web-domain2` do only work for language = en

The translation tool seems to not pick up these modified files and does not translate.

the current :master is broken for me.

```
2024-06-17T08:35:13.116673000Z ERR: node:internal/modules/cjs/loader:1143
2024-06-17T08:35:13.117267000Z   const err = new Error(message);
2024-06-17T08:35:13.117369000Z               ^
2024-06-17T08:35:13.117539000Z Error: Cannot find module 'connect-flash'
2024-06-17T08:35:13.117626000Z Require stack:
2024-06-17T08:35:13.117709000Z - /opt/meshcentral/meshcentral/webserver.js
2024-06-17T08:35:13.117785000Z - /opt/meshcentral/meshcentral/meshcentral.js
2024-06-17T08:35:13.117866000Z     at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
2024-06-17T08:35:13.117946000Z     at Module._load (node:internal/modules/cjs/loader:984:27)
2024-06-17T08:35:13.118024000Z     at Module.require (node:internal/modules/cjs/loader:1231:19)
2024-06-17T08:35:13.118101000Z     at require (node:internal/modules/helpers:179:18)
2024-06-17T08:35:13.118228000Z     at setupDomainAuthStrategy (/opt/meshcentral/meshcentral/webserver.js:7191:21)
2024-06-17T08:35:13.118298000Z     at setupAllDomainAuthStrategies (/opt/meshcentral/meshcentral/webserver.js:6504:88)
2024-06-17T08:35:13.118377000Z     at serverStart (/opt/meshcentral/meshcentral/webserver.js:6491:13)
2024-06-17T08:35:13.118453000Z     at /opt/meshcentral/meshcentral/webserver.js:309:17
2024-06-17T08:35:13.118535000Z     at /opt/meshcentral/meshcentral/db.js:2028:136
2024-06-17T08:35:13.118611000Z     at Query.<anonymous> (/opt/meshcentral/meshcentral/db.js:1301:39) {
2024-06-17T08:35:13.118691000Z   code: 'MODULE_NOT_FOUND',
2024-06-17T08:35:13.118764000Z   requireStack: [
2024-06-17T08:35:13.118844000Z     '/opt/meshcentral/meshcentral/webserver.js',
2024-06-17T08:35:13.118920000Z     '/opt/meshcentral/meshcentral/meshcentral.js'
2024-06-17T08:35:13.118996000Z   ]
2024-06-17T08:35:13.119078000Z }
2024-06-17T08:35:13.119274000Z Node.js v20.12.1
2024-06-17T08:35:13.151954000Z Error: Command failed: /usr/bin/node /opt/meshcentral/meshcentral/meshcentral --configfile config.json --launch 7
2024-06-17T08:35:13.152065000Z node:internal/modules/cjs/loader:1143
2024-06-17T08:35:13.152207000Z   const err = new Error(message);
2024-06-17T08:35:13.152303000Z               ^
2024-06-17T08:35:13.152463000Z Error: Cannot find module 'connect-flash'
2024-06-17T08:35:13.152531000Z Require stack:
2024-06-17T08:35:13.152593000Z - /opt/meshcentral/meshcentral/webserver.js
2024-06-17T08:35:13.152687000Z - /opt/meshcentral/meshcentral/meshcentral.js
2024-06-17T08:35:13.152752000Z     at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
2024-06-17T08:35:13.152852000Z     at Module._load (node:internal/modules/cjs/loader:984:27)

```

--- a/webserver.js
+++ b/webserver.js
@@ -8972,9 +8972,9 @@ module.exports.CreateWebServer = function (parent, db, args, certificates, doneF
         if (obj.renderPages != null) {
             // Get the list of acceptable languages in order
             var acceptLanguages = obj.getLanguageCodes(req);
-
+            var domain = getDomain(req);
             // Take a look at the options we have for this file
-            var fileOptions = obj.renderPages[obj.path.basename(filename)];
+            var fileOptions = obj.renderPages[domain.id][obj.path.basename(filename)];
             if (fileOptions != null) {
                 for (var i in acceptLanguages) {
                     if ((acceptLanguages[i] == 'en') || (acceptLanguages[i].startsWith('en-'))) {
@@ -9021,33 +9021,55 @@ module.exports.CreateWebServer = function (parent, db, args, certificates, doneF
         if (translateFolder != null) {
             obj.renderPages = {};
             obj.renderLanguages = ['en'];
-            var files = obj.fs.readdirSync(translateFolder);
-            for (var i in files) {
-                var name = files[i];
-                if (name.endsWith('.handlebars')) {
-                    name = name.substring(0, name.length - 11);
-                    var xname = name.split('_');
-                    if (xname.length == 2) {
-                        if (obj.renderPages[xname[0]] == null) { obj.renderPages[xname[0]] = {}; }
-                        obj.renderPages[xname[0]][xname[1]] = obj.path.join(translateFolder, name);
-                        if (obj.renderLanguages.indexOf(xname[1]) == -1) { obj.renderLanguages.push(xname[1]); }
-                    }
-                }
-            }
-
-            // See if there are any custom rending pages that will override the default ones
-            if ((obj.parent.webViewsOverridePath != null) && (obj.fs.existsSync(obj.path.join(obj.parent.webViewsOverridePath, 'translations')))) {
-                translateFolder = obj.path.join(obj.parent.webViewsOverridePath, 'translations');
+            for (var i in parent.config.domains) {
+                if (obj.fs.existsSync('views/translations')) { translateFolder = 'views/translations'; }
+                if (obj.fs.existsSync(obj.path.join(__dirname, 'views', 'translations'))) { translateFolder = obj.path.join(__dirname, 'views', 'translations'); }
                 var files = obj.fs.readdirSync(translateFolder);
+                var domain = parent.config.domains[i].id;
+                obj.renderPages[domain] = {};
                 for (var i in files) {
                     var name = files[i];
                     if (name.endsWith('.handlebars')) {
                         name = name.substring(0, name.length - 11);
                         var xname = name.split('_');
                         if (xname.length == 2) {
-                            if (obj.renderPages[xname[0]] == null) { obj.renderPages[xname[0]] = {}; }
-                            obj.renderPages[xname[0]][xname[1]] = obj.path.join(translateFolder, name);
+                            if (obj.renderPages[domain][xname[0]] == null) { obj.renderPages[domain][xname[0]] = {}; }
+                            obj.renderPages[domain][xname[0]][xname[1]] = obj.path.join(translateFolder, name);
                             if (obj.renderLanguages.indexOf(xname[1]) == -1) { obj.renderLanguages.push(xname[1]); }
+                        }
+                    }
+                }
+                // See if there are any custom rending pages that will override the default ones
+                if ((obj.parent.webViewsOverridePath != null) && (obj.fs.existsSync(obj.path.join(obj.parent.webViewsOverridePath, 'translations')))) {
+                    translateFolder = obj.path.join(obj.parent.webViewsOverridePath, 'translations');
+                    var files = obj.fs.readdirSync(translateFolder);
+                    for (var i in files) {
+                        var name = files[i];
+                        if (name.endsWith('.handlebars')) {
+                            name = name.substring(0, name.length - 11);
+                            var xname = name.split('_');
+                            if (xname.length == 2) {
+                                if (obj.renderPages[domain][xname[0]] == null) { obj.renderPages[domain][xname[0]] = {}; }
+                                obj.renderPages[domain][xname[0]][xname[1]] = obj.path.join(translateFolder, name);
+                                if (obj.renderLanguages.indexOf(xname[1]) == -1) { obj.renderLanguages.push(xname[1]); }
+                            }
+                        }
+                    }
+                }
+                // See if there is a custom meshcentral-web-domain folder as that will override the default ones
+                if (obj.fs.existsSync(obj.path.join(__dirname, '..', 'meshcentral-web-' + domain, 'views', 'translations'))) {
+                    translateFolder = obj.path.join(__dirname, '..', 'meshcentral-web-' + domain, 'views', 'translations');
+                    var files = obj.fs.readdirSync(translateFolder);
+                    for (var i in files) {
+                        var name = files[i];
+                        if (name.endsWith('.handlebars')) {
+                            name = name.substring(0, name.length - 11);
+                            var xname = name.split('_');
+                            if (xname.length == 2) {
+                                if (obj.renderPages[domain][xname[0]] == null) { obj.renderPages[domain][xname[0]] = {}; }
+                                obj.renderPages[domain][xname[0]][xname[1]] = obj.path.join(translateFolder, name);
+                                if (obj.renderLanguages.indexOf(xname[1]) == -1) { obj.renderLanguages.push(xname[1]); }
+                            }
                         }
                     }
                 }

